lib/pharos/phases/configure_client.rb

```diff
@@ -1,28 +1,33 @@
 # frozen_string_literal: true
 
-require_relative 'base'
-
 module Pharos
   module Phases
-    class ConfigureClient < Base
-      # @param master [Pharos::Configuration::Host]
-      def initialize(master)
-        @master = master
-        @ssh = Pharos::SSH::Client.for_host(@master)
-      end
+    class ConfigureClient < Pharos::Phase
+      title "Configure kube client"
 
       def call
+        fetch_kubeconfig
+        create_join_command
+      end
+
+      def config_dir
+        File.join(Dir.home, '.pharos')
+      end
+
+      def fetch_kubeconfig
         Dir.mkdir(config_dir, 0o700) unless Dir.exist?(config_dir)
         config_file = File.join(config_dir, @master.address)
+
         logger.info { "Fetching kubectl config ..." }
         config_data = @ssh.file("/etc/kubernetes/admin.conf").read
         File.chmod(0o600, config_file) if File.exist?(config_file)
         File.write(config_file, config_data.gsub(%r{(server: https://)(.+)(:6443)}, "\\1#{@master.address}\\3"), perm: 0o600)
         logger.info { "Configuration saved to #{config_file}" }
       end
 
-      def config_dir
-        File.join(Dir.home, '.pharos')
+      def create_join_command
+        logger.info { "Creating node bootstrap token ..." }
+        @master.join_command = @ssh.exec!("sudo kubeadm token create --print-join-command").split(' ')
```

Comment:
This will now create one bootstrap token on each `pharos-cluster up` run, instead of having `JoinNode` create one for each node during the initial run, but not creating any if the nodes are already joined.

So fewer tokens on the initial `pharos-cluster up` run, but more on later runs.